### ISSUE TITLE
Better tracking options for Ansible package planned work

### ISSUE THREAD

**Issue by samccann** [ansible-community/community-topics#122]
### Summary

Ansible package releases document [roadmaps](https://docs.ansible.com/ansible/devel/roadmap/ansible_roadmap_index.html). This solidifies dates but not planned work.

We could add Planned Work like the core roadmaps do (and update when planned work ends up delayed, etc). This might help at the high level, but we also need better tracking to cover things like when a collection is deprecated and when it's scheduled to be removed to ensure that deprecation and removal work happens on time.

Secondarily, we also have many open [community-topics](https://github.com/ansible-community/community-topics) where we need to solicit votes, especially Steering Committee votes to finalize decisions.  From there, we lack a way to drive/track progress on implementing those decisions.

Let's discuss options here on how we can coordinate and track release deliverables a bit better

**Comment by samccann** [comment]
A specific case - https://github.com/ansible-community/community-topics/issues/79#issuecomment-1195311105

We need a place to track items like this (removal of a deprecated collection in a future release) to ensure we actually followup on that.  Some ideas include:

- Maybe a special todo txt file per future package version in ansible-build-data (maybe todo.7.txt, todo.8.txt, etc) containing short tasks descriptions and links to corresponding issues. Also we could add a check in the build process, e.g. if the file is not empty (or present), fail. 
- Maybe we can create a collection-life-cycle.txt file that tracks ‘supports Ansible x.x.x’, ‘in maintenance’, ‘eol’ details and also make use of GitHub project board!

**Comment by Andersson007** [comment]
@samccann thanks for creating the topic! Whatever we come up with, we should
- conduct a revision of all resolved topics
- find all the things which are planned to happen in Ansible 7, 8, etc.
- add them to our solution:)

**Comment by Andersson007** [comment]
maybe we could create issues (and corresponding milestones) which are associated with releases in the [ansible-build-data](https://github.com/ansible-community/ansible-build-data) repo ?
i don't think we have something else which needs to be done up to certain dates

**Comment by mariolenz** [comment]
Unfortunately, it looks like cross-repository milestones are not possible at the moment: dear-github/dear-github#93

**Comment by mariolenz** [comment]
> maybe we could create issues (and corresponding milestones) which are associated with releases in the [ansible-build-data](https://github.com/ansible-community/ansible-build-data) repo ?

I think it would be worth a try.

@ansible-community/steering-committee What do you think?

**Comment by briantist** [comment]
I use [milestons in `community.hashi_vault`](https://github.com/ansible-collections/community.hashi_vault/milestones) as well and like them a lot.

**Comment by mariolenz** [comment]
> maybe we could create issues (and corresponding milestones) which are associated with releases in the [ansible-build-data](https://github.com/ansible-community/ansible-build-data) repo ?

Is this something we have to discuss further and / or probably vote on, or can someone just add milestones for versions 7 and 8 and we'll try how this works out?

**Comment by felixfontein** [comment]
I think we can just add the milestones. I'll try to do it later today (once I have more stable / faster internet :) ).

**Comment by felixfontein** [comment]
Here they are: https://github.com/ansible-community/ansible-build-data/milestones

**Comment by mariolenz** [comment]
Thanks @felixfontein! Here's a first issue for the [Ansible 8 GA Milestone](https://github.com/ansible-community/ansible-build-data/milestone/2): ansible-community/ansible-build-data#151

Could you please add it?

**Comment by felixfontein** [comment]
@mariolenz done. Though I think we need more milestones, since this is something that should happen once the `8/` directory is created, and not at Ansible 8 GA. I wonder if we ever need the Ansible 8 GA milestone at all, and whether I should simply rename it. If anyone has good ideas for this, please mention them :)

**Comment by mariolenz** [comment]
> Though I think we need more milestones, since this is something that should happen once the `8/` directory is created, and not at Ansible 8 GA.

Yes and no. IMHO, this milestone should collect the things that are planned _for_ the Ansible 8 GA release, not _when_ it's released. So they should already be considered when the `8/` directory is created. The idea is to remove `google.cloud` from the dependencies when the `8/` directory is created in order to prepare the 8 major release. The issue should be closed then, too.

> I wonder if we ever need the Ansible 8 GA milestone at all, and whether I should simply rename it. If anyone has good ideas for this, please mention them :)

Personally, I think having milestones for major releases is very important to track breaking changes. That's the "big picture". Without them, I don't really see milestones for minor releases (or even alphas and betas).

@Andersson007 @briantist @samccann Thoughts?

**Comment by Andersson007** [comment]
>Personally, I think having milestones for major releases is very important to track breaking changes. That's the "big picture". Without them, I don't really see milestones for minor releases (or even alphas and betas).

+1

**Comment by Andersson007** [comment]
I think we should specify in milestones names if they require pre or post actions (i.e. their issues need to be done before or after).
So how about changing them from `Ansible 8 GA` to, say, `Ansible 8 GA Prerequisites` or something?

**Comment by samccann** [comment]
@Andersson007 @felixfontein - where are we on this?  Seems like the milestones worked for Ansible 8.  What would be great is to : 
- create milestones for Ansible 9 (for anything that is either going to be deprecated or removed)
- Find 'some way' to add these to an Ansible 9 roadmap.

The goal is to both ensure we're deprecating/removing on target and to let users know what's coming up in Ansible 9, 10, etc

**Comment by mariolenz** [comment]
@samccann 

>     * create milestones for Ansible 9 (for anything that is either going to be deprecated or removed)

We already have a [milestone](https://github.com/ansible-community/ansible-build-data/milestone/3) for this.

**Comment by samccann** [comment]
Oh excellent! So all we need to do is create an Ansible 9 roadmap and add them!  Here is the issue for that - https://github.com/ansible/ansible/issues/80481

So closing this down as implemented. Thanks everyone!
